apps/emqx_rule_engine/src/emqx_rule_engine_jwt.erl

```diff
@@ -54,3 +76,67 @@ delete_jwt(TId, ResourceId) ->
         error:badarg ->
             ok
     end.
+
+%% @doc Attempts to retrieve a valid JWT from the cache.  If there is
+%% none or if the cached token is expired, generates an caches a fresh
+%% one.
+-spec ensure_jwt(jwt_config()) -> jwt().
+ensure_jwt(JWTConfig) ->
```

Comment:
Just curious: seems that new fun `ensure_jwt` is still called only by jwt worker, while according the desc it is to be called directly somewhere else 🤔

Reply:
In this repo, yes, no other module calls this function.  But in the GCP PubSub rule resource/action, the intention is to call this when publishing messages to generate the authn header, superseding the need for the JWT worker.